server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocator.java

```diff
@@ -325,10 +333,13 @@ private void onNoLongerMaster() {
         }
     }
 
-    private static final class ReconcileDesiredBalanceTask implements ClusterStateTaskListener {
```

Comment:
There seems to be some code convention (not sure if it is strictly followed) for such cluster update tasks to end with `Task`. So I'd be pro keeping it as is.